src/main/scala/mesosphere/mesos/ResourceMatcher.scala

```diff
@@ -236,13 +248,40 @@ object ResourceMatcher extends StrictLogging {
       } else true
     }
 
+    val checkGpuSchedulingBehaviour: Boolean = {
+      val availableGPUs = groupedResources.getOrElse(Resource.GPUS, Nil).foldLeft(0.0)(_ + _.getScalar.getValue)
+      val gpuResourcesAreWasted = availableGPUs > 0 && runSpec.resources.gpus == 0
+      conf.gpuSchedulingBehavior() match {
+        case GpuSchedulingBehavior.Undefined =>
+          if (gpuResourcesAreWasted) {
+            addOnMatch(() => logger.warn(s"Runspec [${runSpec.id}] doesn't require any GPU resources but " +
+              "will be launched on an agent with GPU resources."))
+          }
+          true
+
+        case GpuSchedulingBehavior.Restricted =>
+          val noPersistentVolumeToMatch = PersistentVolumeMatcher.matchVolumes(offer, reservedInstances).isEmpty
+          if (gpuResourcesAreWasted && noPersistentVolumeToMatch) {
+            noOfferMatchReasons += NoOfferMatchReason.DeclinedScarceResources
+            false
+          } else {
+            true
```

Comment:
should we maybe log something here? That we are launching on GPU node while restricted but we are doing that because of persistent volume?

Reply:
Design doc says nothing about it. @timcharper what do you think?

Reply:
I did not mean warn or error, I would log just info... But @timcharper feel free to comment :)